<?xml version="1.0" encoding="utf-8"?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from fake_car.urdf.xacro            | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="fake_car" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <!--
  <macro name="cylinder_inertial" params="m r h">
    <inertial>
      <mass value="${m}"/>
      <xacro:cylinder_inertia m="${m}" r="${r}" h="${h}"/>
    </inertial>
  </macro>
-->
  <link name="base_link">
    <visual>
      <geometry>
        <cylinder length="0.02" radius="0.01"/>
      </geometry>
      <material name="silver">
        <color rgba="0.75 0.75 0.75 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.02" radius="0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <inertia ixx="0.015" ixy="0" ixz="0" iyy="0.0375" iyz="0" izz="0.0375"/>
    </inertial>
  </link>
  <link name="back_left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.047" radius="0.056"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.047" radius="0.056"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1.0"/>
      <inertia ixx="0.015" ixy="0" ixz="0" iyy="0.0375" iyz="0" izz="0.0375"/>
    </inertial>
    <!-- <xacro:cylinder_inertial m="${wheel_mass}" r="${wheel_radius}" h="${wheel_width}"/> -->
  </link>
  <link name="laser_scanner_link">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size=".127 0.1016 0.0508"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size=".127 0.1016 0.0508"/>
      </geometry>
    </visual>
    <inertial>
      <mass value="1e-5"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="1e-6" ixy="0" ixz="0" iyy="1e-6" iyz="0" izz="1e-6"/>
    </inertial>
  </link>
  <joint name="laser_scanner_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.1397 0 0.2159"/>
    <parent link="base_link"/>
    <child link="laser_scanner_link"/>
  </joint>
  <gazebo reference="laser_scanner_link">
    <sensor name="laser_scanner" type="gpu_ray">
      <pose>0 0 0 0 0 0</pose>
      <visualize>true</visualize>
      <update_rate>40</update_rate>
      <ray>
        <scan>
          <horizontal>
            <samples>360</samples>
            <resolution>1</resolution>
            <min_angle>-3.14159265359</min_angle>
            <max_angle>3.14159265359</max_angle>
          </horizontal>
        </scan>
        <range>
          <min>0.10</min>
          <max>15.0</max>
          <resolution>0.1</resolution>
        </range>
      </ray>
      <plugin filename="libgazebo_ros_gpu_laser.so" name="gpu_laser">
        <topicName>/scan</topicName>
        <frameName>laser_scanner_link</frameName>
      </plugin>
    </sensor>
  </gazebo>
  <link name="back_right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.047" radius="0.056"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.047" radius="0.056"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1"/>
      <inertia ixx="5.1458e-5" ixy="0" ixz="0" iyy="5.1458e-5" iyz="0" izz="6.125e-5"/>
    </inertial>
  </link>
  <link name="front_left_wheel_steer">
    <visual>
      <geometry>
        <cylinder length="0.01" radius="0.01"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.01" radius="0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1"/>
      <inertia ixx="5.1458e-5" ixy="0" ixz="0" iyy="5.1458e-5" iyz="0" izz="6.125e-5"/>
    </inertial>
  </link>
  <link name="front_left_wheel">
    <visual>
      <geometry>
        <cylinder length="0.047" radius="0.056"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.047" radius="0.056"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1"/>
      <inertia ixx="5.1458e-5" ixy="0" ixz="0" iyy="5.1458e-5" iyz="0" izz="6.125e-5"/>
    </inertial>
  </link>
  <link name="front_right_wheel_steer">
    <visual>
      <geometry>
        <cylinder length="0.01" radius="0.01"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.01" radius="0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1"/>
      <inertia ixx="5.1458e-5" ixy="0" ixz="0" iyy="5.1458e-5" iyz="0" izz="6.125e-5"/>
    </inertial>
  </link>
  <link name="front_right_wheel">
    <visual>
      <geometry>
        <cylinder length="0.047" radius="0.056"/>
      </geometry>
      <material name="black">
        <color rgba="0 0 0 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <cylinder length="0.047" radius="0.056"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="0.1"/>
      <inertia ixx="5.1458e-5" ixy="0" ixz="0" iyy="5.1458e-5" iyz="0" izz="6.125e-5"/>
    </inertial>
  </link>
  <joint name="back_left_wheel_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="back_left_wheel"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0.125 0"/>
    <limit effort="10000" velocity="500"/>
  </joint>
  <joint name="back_right_wheel_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="back_right_wheel"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0.125 0.0"/>
    <limit effort="10000" velocity="500"/>
  </joint>
  <joint name="front_left_wheel_steer_joint" type="revolute">
    <limit effort="1" lower="-1" upper="1" velocity="5"/>
    <axis xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="front_left_wheel_steer"/>
    <origin rpy="0 0 0" xyz="0.3429 0.125 0"/>
  </joint>
  <joint name="front_left_wheel_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="front_left_wheel_steer"/>
    <child link="front_left_wheel"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
  </joint>
  <joint name="front_right_wheel_steer_joint" type="revolute">
    <limit effort="1" lower="-1" upper="1" velocity="5"/>
    <axis xyz="0 0 1"/>
    <parent link="base_link"/>
    <child link="front_right_wheel_steer"/>
    <origin rpy="0 0 0" xyz="0.3429 -0.125 0"/>
  </joint>
  <joint name="front_right_wheel_joint" type="continuous">
    <axis xyz="0 0 1"/>
    <parent link="front_right_wheel_steer"/>
    <child link="front_right_wheel"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
  </joint>
  <gazebo reference="front_left_wheel">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="front_right_wheel">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="back_left_wheel">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="back_right_wheel">
    <material>Gazebo/DarkGrey</material>
  </gazebo>
  <gazebo reference="base_link">
    <dampingFactor>0.00007</dampingFactor>
  </gazebo>
  <gazebo>
    <plugin filename="libfake_car_plugin.so" name="fake_car_plugin">
    </plugin>
  </gazebo>
</robot>

